File: frontend/src/model/photo.ts
import { Location } from 'model/location';
import { CameraGear, CameraSettings } from 'model/camera';
import { Metadata } from 'model/metadata';

export type Judgement = {
  rating: number;
  inShowroom: boolean;
};

export type Photo = {
  imageId: string;
  photographer: string;
  title: string;
  description?: string;
  group?: string;
  taken: Date;
  location: Location;
  gear: CameraGear;
  cameraSettings: CameraSettings;
  metadata: Metadata;
  judgement: Judgement;
};

File: backend/photography/entities/src/main/scala/backend/photography/entities/photo/Photo.scala
package backend.photography.entities.photo

import backend.photography.entities.photo.gear.Gear
import backend.photography.entities.photo.meta.{Judgement, Location, Metadata}
import backend.photography.entities.photo.settings.CameraSettings

import java.time.Instant

case class Photo(imageId: String,
                 title: String,
                 description: Option[String],
                 photographer: String,
                 group: Option[String],
                 location: Location,
                 taken: Instant,
                 gear: Gear,
                 cameraSettings: CameraSettings,
                 metadata: Metadata,
                 judgement: Judgement)
